### ISSUE TITLE
Checkout dependency repo on delete action

### ISSUE THREAD

**Issue by kayrus** [gardener/sow#19]
When the repo is not checked out, e.g. gen directory is empty, sow fails on delete action:

```
*** deleting species cert-manager/controller
Plugin: kubectl[delete] (/sow/plugins/kubectl/plugin)
manifest for Issuer(certmanager.k8s.io/v1alpha1): garden/acme-issuer
cluster https://garden-root
  found deployed objects: 1
  found currect objects: 0
  found deployed obsolete objects: 1
    Issuer(certmanager.k8s.io/v1alpha1): garden/acme-issuer
issuer.certmanager.k8s.io "acme-issuer" deleted
Using ACME server at https://acme-staging-v02.api.letsencrypt.org/directory
Plugin: helm[delete] (/sow/plugins/helm/plugin)
    using /mounted/dev/kubeconfig_sa
Error: helm source 'chart-checkout/charts/cert-manager' not found
```

**Comment by kayrus** [comment]
Faced this issue again. Tried to checkout the repo using `sow deploy --dryrun`, but this now causes:

```
sow delete gardener/virtual
*****************************
*** deleting species gardener/virtual
deployment has changed
Plugin: kubectl[delete] (/sow/plugins/kubectl/plugin)
manifest for Secret(v1): garden/garden-secret-runtime
Plugin: kubectl[delete] (/sow/plugins/kubectl/plugin)
Error: entry 0:0: gardener/rendered_charts.yaml not found
```

**Comment by kayrus** [comment]
Another case:

```
*** deleting species etcd/backupinfra
Plugin: terraform[delete] (/sow/plugins/terraform/plugin)
Plugin reinitialization required. Please run "terraform init".
```

**Comment by Diaphteiros** [comment]
This has been solved with sow version `1.1.1` and garden-setup `1.13.0`.
The `git`, `terraform`, and `helm` plugins of sow were changed to perform some actions they only did on deploy now also on delete, to create files necessary for other plugins (etc. `kubectl`). 
In garden-setup, the order of the plugins has been modified to make sure the plugins downloading/creating manifests for other plugins are always called before these other plugins (the order of components is usually reverted by sow on a delete action).

With these changes, the `gen` folder is not needed for deletion of a landscape anymore. `state` and `export` are still needed, although the latter one can be recreated by deploying everything again.
